Functions/CatalogItems/Out-RsCatalogItem.ps1

```diff
@@ -1,129 +1,184 @@
+            {
+                return '.rsds'
+            }
+            elseif ($TypeName -eq 'DataSet')
+            {
+                return '.rsd'
+            }
+            else
+            {
+                throw 'Unsupported item type! We only support items which are of type Report, Data Set or Data Source'
+            }
         }
-        elseif ($TypeName -eq 'DataSource') 
-        {
-            return '.rsds'
-        }
-        elseif ($TypeName -eq 'DataSet')
-        {
-            return '.rsd'
-        } 
-        else 
+        #endregion Utility Functions
+        
+        #region Connect to Report Server using Web Proxy
+        if (-not $Proxy)
         {
-            throw 'Unsupported item type! We only support items which are of type Report, Data Set or Data Source'
+            try
+            {
+                $splat = @{ }
+                if ($PSBoundParameters.ContainsKey('ReportServerUri')) { $splat['ReportServerUri'] = $ReportServerUri }
+                if ($PSBoundParameters.ContainsKey('Credential')) { $splat['Credential'] = $Credential }
+                $Proxy = New-RSWebServiceProxy @splat
+            }
+            catch
+            {
+                throw
+            }
         }
+        #endregion Connect to Report Server using Web Proxy
+        
+        $DestinationFullPath = Resolve-Path $Destination
     }
-
-    if (-not $Proxy)
-    {
-        $Proxy = New-RSWebServiceProxy -ReportServerUri $ReportServerUri -Credentials $ReportServerCredentials 
-    }
-
-    $itemType = $Proxy.GetItemType($Path)
-    if ($itemType -eq 'Unknown')
-    {
-        throw "Make sure item exists at $Path and item is of type Report, DataSet, DataSource or Resource"
-    }
-    elseif ($itemType -eq 'Resource')
-    {
-        $itemName = ($path.Split("/"))[-1]
-        # Resource contain the file extension as part of their name, so we don't need to call Get-FileExtension
-        $fileName = $itemName
-    }
-    else 
-    {
-        $itemName = ($path.Split("/"))[-1]
-        $fileName = $itemName + (Get-FileExtension $itemType)
-    }
-
-    Write-Verbose "Downloading $Path..."
-    $bytes = $Proxy.GetItemDefinition($Path)
     
-    if (!(Test-Path -Path $Destination))
+    Process
     {
-        Write-Verbose "Creating Folder $Destination..."
-        New-Item -ItemType directory -Path $Destination | Out-Null
-    }
-
-    $DestinationFullPath = Resolve-Path $Destination
-    Write-Verbose "Writing $itemType content to $DestinationFullPath\$fileName..."
-    if ($itemType -eq 'DataSource')
-    {
-        $content = [System.Text.Encoding]::Unicode.GetString($bytes)
-        [System.IO.File]::WriteAllText("$DestinationFullPath\$fileName", $content)
+        #region Processing each path passed to it
+        foreach ($item in $Path)
+        {
+            #region Retrieving content from Report Server
+            try { $itemType = $Proxy.GetItemType($item) }
```

Comment:
In general please use  a different line for the brackets in the try catch scenarios even for single line